### PR TITLE
Add merger mode for HIDDMXDevices (DE/FX5/Nodle)

### PR DESCRIPTION
**Pull request by qfulmina**
Adding the ability to set the DE/FX5/Nodle devices into merger mode as explained in the updated docs.

Tested with two Nodle R4S.

### CODE REVIEW

File: plugins/hid/hiddmxdevice.h [hunk 1]
```diff
@@ -117,7 +129,8 @@ class HIDDMXDevice : public HIDDevice
     {
```
Comment (qfulmina) [inline plugins/hid/hiddmxdevice.h:129]: Here I went for minimal change. We could also reflect the devices' internal bitmap here. That would be a change to:

```
    {
        DMX_MODE_NONE   = 1 << 0,
        DMX_MODE_MERGER = 1 << 1,
        DMX_MODE_OUTPUT  = 1 << 2,
        DMX_MODE_INPUT  = 1 << 3
    };
```

This way `m_mode` could be written to the device as is, saving us from fiddling around with ifs in [#plugins/hid/hiddmxdevice.cpp](https://github.com/mcallegari/qlcplus/blob/b8d7cb10dafb24159ea6ba2e21524b798ec48eb0/plugins/hid/hiddmxdevice.cpp#L227C1-L232C26)

I could rework this part to make it cleaner. Just now it's "less intrusive".

File: plugins/hid/configurehid.cpp [hunk 2]
```diff
@@ -100,6 +103,12 @@ void ConfigureHID::refreshList()
         item->setText(KColumnNumber, s.setNum(i + 1));
         item->setText(KColumnName, dev->name());
         item->setFlags(item->flags() | Qt::ItemIsUserCheckable);
+
+        if (dev->hasMergerMode()) {
```
Comment (mcallegari) [inline plugins/hid/configurehid.cpp:107]: Please stick to the project's indentation. Parenthesis on new line

File: plugins/hid/configurehid.cpp [hunk 3]
```diff
@@ -124,3 +133,32 @@ void ConfigureHID::slotDeviceRemoved(HIDDevice* device)
         }
     }
 }
+
+QWidget* ConfigureHID::createMergerModeWidget(bool mergerModeEnabled)
+{
+    QCheckBox* checkbox = new QCheckBox;
+
+        if (mergerModeEnabled)
```
Comment (mcallegari) [inline plugins/hid/configurehid.cpp:141]: Please review indentation.

File: plugins/hid/hiddmxdevice.cpp [hunk 4]
```diff
@@ -75,6 +75,18 @@ void HIDDMXDevice::init()
...
  *****************************************************************************/
 
+bool HIDDMXDevice::isMergerModeEnabled()
+{
+    return (m_mode & DMX_MODE_MERGER);
+}
+
+void HIDDMXDevice::enableMergerMode(bool mergerModeEnabled)
+{
+    if (mergerModeEnabled) m_mode |= DMX_MODE_MERGER;
```
Comment (mcallegari) [inline plugins/hid/hiddmxdevice.cpp:85]: Please review indentation. code on new line after condition

File: plugins/hid/hiddmxdevice.h [hunk 5]
```diff
@@ -64,10 +64,22 @@ class HIDDMXDevice : public HIDDevice
     /** @reimp */
     bool hasOutput() { return true; }
 
+    /** @reimp */
+    bool hasMergerMode() {
```
Comment (mcallegari) [inline plugins/hid/hiddmxdevice.h:68]: Please stick to the project's indentation. Parenthesis on new line

File: plugins/hid/hiddmxdevice.h [hunk 6]
```diff
@@ -117,7 +129,8 @@ class HIDDMXDevice : public HIDDevice
     {
         DMX_MODE_NONE   = 1 << 0,
         DMX_MODE_OUTPUT = 1 << 1,
-        DMX_MODE_INPUT  = 1 << 2
+        DMX_MODE_INPUT  = 1 << 2,
+        DMX_MODE_MERGER  = 1 << 3
```
Comment (mcallegari) [inline plugins/hid/hiddmxdevice.h:133]: double space -> single space

**Comment by mcallegari** [comment]
Merged, thanks a lot 👍

**Comment by qfulmina** [comment]
Thank _you_!

**Pull request merged**